### ISSUE TITLE
`Cannot cast object 'null' with class 'null' to class 'long'. Try 'java.lang.Long' instead` in NF 0.27.0

### ISSUE THREAD

**Issue by mypandos** [nextflow-io/nextflow#575]
```Jan-18 10:20:57.513 [Actor Thread 3] ERROR nextflow.Session - Cannot cast object 'null' with class 'null' to class 'long'. Try 'java.lang.Long' instead
org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'null' with class 'null' to class 'long'. Try 'java.lang.Long' instead
        at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.castToNumber(DefaultTypeTransformation.java:163)
        at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.longUnbox(DefaultTypeTransformation.java:78)
        at nextflow.trace.WorkflowStats.getCpuTime(WorkflowStats.groovy:137)
        at nextflow.trace.WorkflowStats.updateTasksCached(WorkflowStats.groovy:172)
        at nextflow.trace.StatsObserver.onProcessCached(StatsObserver.groovy:80)
        at nextflow.Session.notifyTaskCached(Session.groovy:831)
        at nextflow.Session$notifyTaskCached$1.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at nextflow.processor.TaskProcessor.checkCachedOutput(TaskProcessor.groovy:995)
        at nextflow.processor.TaskProcessor$checkCachedOutput$2.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
        at nextflow.processor.TaskProcessor.checkCachedOrLaunchTask(TaskProcessor.groovy:841)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
        at nextflow.processor.TaskProcessor.invokeTask(TaskProcessor.groovy:674)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSite.invoke(PogoMetaMethodSite.java:169)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at nextflow.processor.TaskProcessor$InvokeTaskAdapter.call(TaskProcessor.groovy:225)
        at groovyx.gpars.dataflow.operator.DataflowOperatorActor.startTask(DataflowOperatorActor.java:120)
        at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor.access$001(ForkingDataflowOperatorActor.java:35)
        at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor$1.run(ForkingDataflowOperatorActor.java:58)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by pditommaso** [comment]
I see, thanks for reporting that. I will upload a patch soon. In the meanwhile you can avoid this issue turning off the html report feature.

**Comment by pditommaso** [comment]
I've just uploaded a patch fixing this problem. Please confirm that works fine running 

```
NXF_VER=0.27.1-SNAPSHOT nextflow run .. your command line 
```

(the first time you run it will download the updated libraries)

**Comment by mypandos** [comment]
The error is still there, even with turning off html report feature. Full log below:
```
Jan-18 11:32:37.239 [main] DEBUG nextflow.cli.Launcher - $> /home/mamana/.conda/envs/imputation/bin/nextflow -log /researchdata/fhgfs/mamana/imputation/LOG/imputation_nf.log run /home/mamana/imputation/chipimputation/imputation.nf -c /home/mamana/imputation/chipimputation/imputation_nf.config -w /researchdata/fhgfs/mamana/imputation/work -resume -profile pbs
Jan-18 11:32:37.775 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 0.27.1-SNAPSHOT
Jan-18 11:32:37.806 [main] INFO  nextflow.cli.CmdRun - Launching `/home/mamana/imputation/chipimputation/imputation.nf` [mad_booth] - revision: 2108e0700f
Jan-18 11:32:37.844 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/mamana/imputation/chipimputation/imputation_nf.config
Jan-18 11:32:37.849 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/mamana/imputation/chipimputation/imputation_nf.config
Jan-18 11:32:38.395 [main] DEBUG nextflow.config.ConfigBuilder - Setting config profile: 'pbs'
Jan-18 11:32:38.757 [main] DEBUG nextflow.Session - Session uuid: c1f944ca-83cb-4272-9587-3f964b255efc
Jan-18 11:32:38.757 [main] DEBUG nextflow.Session - Run name: mad_booth
Jan-18 11:32:38.762 [main] DEBUG nextflow.Session - Executor pool size: 64
Jan-18 11:32:38.789 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 0.27.1-SNAPSHOT build 4752
  Modified: 18-01-2018 08:54 UTC (10:54 SAST)
  System: Linux 3.0.101-108.10-default
  Runtime: Groovy 2.4.13 on OpenJDK 64-Bit Server VM 1.8.0_121-b15
  Encoding: UTF-8 (UTF-8)
  Process: 8468@srvslshpc600.uct.ac.za [137.158.155.219]
  CPUs: 64 - Mem: 126 GB (31.2 GB) - Swap: 0 (0)
Jan-18 11:32:39.007 [main] DEBUG nextflow.Session - Work-dir: /researchdata/fhgfs/mamana/imputation/work [fhgfs]
Jan-18 11:32:39.008 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /home/mamana/imputation/chipimputation/bin
Jan-18 11:32:39.263 [main] DEBUG nextflow.Session - Session start invoked
Jan-18 11:32:39.270 [main] DEBUG nextflow.processor.TaskDispatcher - Dispatcher > start
Jan-18 11:32:39.271 [main] DEBUG nextflow.script.ScriptRunner - > Script parsing
Jan-18 11:32:39.955 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Jan-18 11:32:40.248 [main] DEBUG nextflow.processor.ProcessFactory - << taskConfig executor: pbs
Jan-18 11:32:40.249 [main] DEBUG nextflow.processor.ProcessFactory - >> processorType: 'pbs'
Jan-18 11:32:40.268 [main] DEBUG nextflow.executor.Executor - Initializing executor: pbs
Jan-18 11:32:40.273 [main] INFO  nextflow.executor.Executor - [warm up] executor > pbs
Jan-18 11:32:40.284 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'pbs' > capacity: 100; pollInterval: 5s; dumpInterval: 5m 
Jan-18 11:32:40.288 [main] DEBUG nextflow.processor.TaskDispatcher - Starting monitor: TaskPollingMonitor
Jan-18 11:32:40.290 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: pbs)
Jan-18 11:32:40.295 [main] DEBUG nextflow.executor.Executor - Invoke register for executor: pbs
Jan-18 11:32:40.296 [main] DEBUG n.executor.AbstractGridExecutor - Creating executor 'pbs' > queue-stat-interval: 1m
Jan-18 11:32:40.359 [main] DEBUG nextflow.Session - >>> barrier register (process: check_chromosome)
Jan-18 11:32:40.366 [main] DEBUG nextflow.processor.TaskProcessor - Creating operator > check_chromosome -- maxForks: 64
Jan-18 11:32:40.677 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [7d/31709f] Cached process > check_chromosome (check_chromosome_tiny.yri)
Jan-18 11:32:40.691 [Actor Thread 3] ERROR nextflow.Session - Cannot cast object 'null' with class 'null' to class 'int'. Try 'java.lang.Integer' instead
org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'null' with class 'null' to class 'int'. Try 'java.lang.Integer' instead
	at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.castToNumber(DefaultTypeTransformation.java:163)
	at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.intUnbox(DefaultTypeTransformation.java:69)
	at nextflow.trace.WorkflowStats.getCpuTime(WorkflowStats.groovy:137)
	at nextflow.trace.WorkflowStats.updateTasksCached(WorkflowStats.groovy:172)
	at nextflow.trace.StatsObserver.onProcessCached(StatsObserver.groovy:80)
	at nextflow.Session.notifyTaskCached(Session.groovy:831)
	at nextflow.Session$notifyTaskCached$1.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.processor.TaskProcessor.checkCachedOutput(TaskProcessor.groovy:995)
	at nextflow.processor.TaskProcessor$checkCachedOutput$2.callCurrent(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
	at nextflow.processor.TaskProcessor.checkCachedOrLaunchTask(TaskProcessor.groovy:841)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
	at nextflow.processor.TaskProcessor.invokeTask(TaskProcessor.groovy:674)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSite.invoke(PogoMetaMethodSite.java:169)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.processor.TaskProcessor$InvokeTaskAdapter.call(TaskProcessor.groovy:225)
	at groovyx.gpars.dataflow.operator.DataflowOperatorActor.startTask(DataflowOperatorActor.java:120)
	at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor.access$001(ForkingDataflowOperatorActor.java:35)
	at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor$1.run(ForkingDataflowOperatorActor.java:58)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Jan-18 11:32:40.743 [Actor Thread 2] DEBUG nextflow.Session - <<< barrier arrive (process: check_chromosome)
Jan-18 11:32:40.760 [main] DEBUG nextflow.processor.ProcessFactory - << taskConfig executor: pbs
Jan-18 11:32:40.761 [main] DEBUG nextflow.processor.ProcessFactory - >> processorType: 'pbs'
Jan-18 11:32:40.762 [main] DEBUG nextflow.executor.Executor - Initializing executor: pbs
Jan-18 11:32:40.763 [main] DEBUG n.executor.AbstractGridExecutor - Creating executor 'pbs' > queue-stat-interval: 1m
Jan-18 11:32:40.765 [main] DEBUG nextflow.Session - >>> barrier register (process: generate_chunks)
Jan-18 11:32:40.766 [main] DEBUG nextflow.processor.TaskProcessor - Creating operator > generate_chunks -- maxForks: 64
Jan-18 11:32:40.825 [main] DEBUG nextflow.processor.ProcessFactory - << taskConfig executor: pbs
Jan-18 11:32:40.825 [main] DEBUG nextflow.processor.ProcessFactory - >> processorType: 'pbs'
Jan-18 11:32:40.828 [main] DEBUG nextflow.executor.Executor - Initializing executor: pbs
Jan-18 11:32:40.829 [main] DEBUG n.executor.AbstractGridExecutor - Creating executor 'pbs' > queue-stat-interval: 1m
Jan-18 11:32:40.833 [main] DEBUG nextflow.Session - >>> barrier register (process: plink_to_chrm)
Jan-18 11:32:40.837 [main] DEBUG nextflow.processor.TaskProcessor - Creating *combiner* operator for each param(s) at index(es): [0]
Jan-18 11:32:40.846 [Actor Thread 1] INFO  nextflow.processor.TaskProcessor - [cf/2f2013] Cached process > generate_chunks (generate_chunks_tiny.yri)
Jan-18 11:32:40.854 [main] DEBUG nextflow.processor.TaskProcessor - Creating operator > plink_to_chrm -- maxForks: 64
Jan-18 11:32:40.857 [Actor Thread 2] DEBUG nextflow.Session - <<< barrier arrive (process: generate_chunks)
Jan-18 11:32:40.866 [main] DEBUG nextflow.script.ScriptRunner - > Await termination 
Jan-18 11:32:40.867 [main] DEBUG nextflow.Session - Session await
Jan-18 11:32:40.908 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [96/7f8d1b] Cached process > plink_to_chrm (plink2chrm_22_tiny.yri)
Jan-18 11:32:40.917 [Actor Thread 2] DEBUG nextflow.Session - <<< barrier arrive (process: plink_to_chrm)
Jan-18 11:32:40.918 [main] DEBUG nextflow.Session - Session await > all process finished
Jan-18 11:32:45.314 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: pbs)
Jan-18 11:32:45.316 [main] DEBUG nextflow.Session - Session await > all barriers passed
Jan-18 11:32:45.346 [main] DEBUG nextflow.trace.StatsObserver - Workflow completed > WorkflowStats[succeedCount=0; failedCount=0; ignoredCount=0; cachedCount=2; succeedDuration=0ms; failedDuration=0ms; cachedDuration=1s]
Jan-18 11:32:45.371 [main] DEBUG nextflow.CacheDB - Closing CacheDB done
Jan-18 11:32:45.461 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye

```

**Comment by pditommaso** [comment]
Ok, it was a bit tricky. Now it's fixed. If you want to try it run: 

```
CAPSULE_RESET=1 NXF_VER=0.27.1-SNAPSHOT nextflow info
```

to update your snapshot, then: 

```
NXF_VER=0.27.1-SNAPSHOT nextflow run .. <your command line>
```

**Comment by mypandos** [comment]
Fixed! Great.

**Comment by pditommaso** [comment]
Good. Included in version `0.27.1` just released.
